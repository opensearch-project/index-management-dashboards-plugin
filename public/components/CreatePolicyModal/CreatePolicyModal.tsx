/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from "react";
import {
  EuiButton,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiButtonEmpty,
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiFormRow,
  EuiRadio,
  EuiPanel,
} from "@elastic/eui";

interface CreatePolicyModalProps {
  isEdit?: boolean;
  onClose: () => void;
  onClickContinue: (visual: boolean) => void;
}

const CreatePolicyModal: React.FC<CreatePolicyModalProps> = ({ isEdit = false, onClose, onClickContinue }) => {
  const [visual, setVisual] = useState(true);
  return (
    <EuiOverlayMask>
      {/*
      // @ts-ignore */}
      <EuiModal onCancel={onClose} onClose={onClose} maxWidth={600}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>Configuration method</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiFlexGroup gutterSize="m" direction="column" style={{ margin: "-4px" }}>
            <EuiFlexItem grow={false}>
              <EuiText size="s" style={{ marginTop: 0 }}>
                Choose how you would like to {isEdit ? "modify" : "define"} your policy, either using a visual editor or writing JSON.
              </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiFlexGroup>
                <EuiFlexItem>
                  <EuiPanel className={visual ? "selected-radio-panel" : ""}>
                    <EuiFormRow
                      helpText={`Use the visual editor to ${isEdit ? "update" : "create"} your policy${
                        isEdit ? "" : " using pre-defined options."
                      }`}
                    >
                      <EuiRadio
                        id="create-policy-visual"
                        label="Visual editor"
                        checked={visual}
                        onChange={(e) => setVisual(e.target.checked)}
                        data-test-subj="createPolicyModalVisualRadio"
                      />
                    </EuiFormRow>
                  </EuiPanel>
                </EuiFlexItem>
                <EuiFlexItem>
                  <EuiPanel className={visual ? "" : "selected-radio-panel"}>
                    <EuiFormRow helpText={`Use the JSON editor to ${isEdit ? "update" : "create or import"} your policy using JSON.`}>
                      <EuiRadio
                        id="create-policy-json"
                        label="JSON editor"
                        checked={!visual}
                        onChange={(e) => setVisual(!e.target.checked)}
                        data-test-subj="createPolicyModalJsonRadio"
                      />
                    </EuiFormRow>
                  </EuiPanel>
                </EuiFlexItem>
              </EuiFlexGroup>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiFlexGroup justifyContent="flexEnd">
            <EuiFlexItem grow={false}>
              <EuiButtonEmpty onClick={onClose} data-test-subj="createPolicyModalCancelButton">
                Cancel
              </EuiButtonEmpty>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiButton
                onClick={() => {
                  onClose();
                  onClickContinue(visual);
                }}
                fill
                data-test-subj="createPolicyModalContinueButton"
              >
                Continue
              </EuiButton>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};

export default CreatePolicyModal;
