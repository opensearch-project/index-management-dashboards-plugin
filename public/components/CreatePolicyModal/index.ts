/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import CreatePolicyModal from "./CreatePolicyModal";

export default CreatePolicyModal;
