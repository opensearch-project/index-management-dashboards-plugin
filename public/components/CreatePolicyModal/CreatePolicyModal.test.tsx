/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";
import CreatePolicyModal from "./CreatePolicyModal";

describe("<CreatePolicyModal /> spec", () => {
  it("renders the component", () => {
    render(<CreatePolicyModal isEdit={false} onClose={() => {}} onClickContinue={() => {}} />);
    // EuiOverlayMask appends an element to the body so we should have two, an empty div from react-test-library
    // and our EuiOverlayMask element
    expect(document.body.children).toHaveLength(2);
    expect(document.body.children[1]).toMatchSnapshot();
  });

  it("renders the component w/ edit", () => {
    render(<CreatePolicyModal isEdit={true} onClose={() => {}} onClickContinue={() => {}} />);
    // EuiOverlayMask appends an element to the body so we should have two, an empty div from react-test-library
    // and our EuiOverlayMask element
    expect(document.body.children).toHaveLength(2);
    expect(document.body.children[1]).toMatchSnapshot();
  });

  it("calls onAction and onCLose when action button clicked", () => {
    const onContinue = jest.fn();
    const onClose = jest.fn();
    const { getByTestId } = render(<CreatePolicyModal isEdit={false} onClose={onClose} onClickContinue={onContinue} />);

    fireEvent.click(getByTestId("createPolicyModalContinueButton"));
    expect(onContinue).toHaveBeenCalled();
    expect(onClose).toHaveBeenCalled();
  });

  it("calls close when close button clicked", () => {
    const onClose = jest.fn();
    const { getByTestId } = render(<CreatePolicyModal isEdit={false} onClose={onClose} onClickContinue={() => {}} />);

    fireEvent.click(getByTestId("createPolicyModalCancelButton"));
    expect(onClose).toHaveBeenCalled();
  });
});
