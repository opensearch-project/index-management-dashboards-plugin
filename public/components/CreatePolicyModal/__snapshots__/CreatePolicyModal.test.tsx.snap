// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePolicyModal /> spec renders the component 1`] = `
<div
  class="euiOverlayMask euiOverlayMask--aboveHeader"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal"
      style="max-width: 600px;"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
        type="button"
      >
        EuiIconMock
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            Configuration method
          </div>
        </div>
        <div
          class="euiModalBody"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--directionColumn euiFlexGroup--responsive"
              style="margin: -4px;"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiText euiText--small"
                  style="margin-top: 0px;"
                >
                  Choose how you would like to 
                  define
                   your policy, either using a visual editor or writing JSON.
                </div>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <div
                    class="euiFlexItem"
                  >
                    <div
                      class="euiPanel euiPanel--paddingMedium selected-radio-panel"
                    >
                      <div
                        class="euiFormRow"
                        id="some_html_id-row"
                      >
                        <div
                          class="euiFormRow__fieldWrapper"
                        >
                          <div
                            aria-describedby="some_html_id-help"
                            class="euiRadio"
                            data-test-subj="createPolicyModalVisualRadio"
                          >
                            <input
                              checked=""
                              class="euiRadio__input"
                              id="some_html_id"
                              type="radio"
                              value=""
                            />
                            <div
                              class="euiRadio__circle"
                            />
                            <label
                              class="euiRadio__label"
                              for="some_html_id"
                            >
                              Visual editor
                            </label>
                          </div>
                          <div
                            class="euiFormHelpText euiFormRow__text"
                            id="some_html_id-help"
                          >
                            Use the visual editor to create your policy using pre-defined options.
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiFlexItem"
                  >
                    <div
                      class="euiPanel euiPanel--paddingMedium"
                    >
                      <div
                        class="euiFormRow"
                        id="some_html_id-row"
                      >
                        <div
                          class="euiFormRow__fieldWrapper"
                        >
                          <div
                            aria-describedby="some_html_id-help"
                            class="euiRadio"
                            data-test-subj="createPolicyModalJsonRadio"
                          >
                            <input
                              class="euiRadio__input"
                              id="some_html_id"
                              type="radio"
                              value=""
                            />
                            <div
                              class="euiRadio__circle"
                            />
                            <label
                              class="euiRadio__label"
                              for="some_html_id"
                            >
                              JSON editor
                            </label>
                          </div>
                          <div
                            class="euiFormHelpText euiFormRow__text"
                            id="some_html_id-help"
                          >
                            Use the JSON editor to create or import your policy using JSON.
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--primary"
                data-test-subj="createPolicyModalCancelButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonEmpty__content"
                >
                  <span
                    class="euiButtonEmpty__text"
                  >
                    Cancel
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary euiButton--fill"
                data-test-subj="createPolicyModalContinueButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Continue
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;

exports[`<CreatePolicyModal /> spec renders the component w/ edit 1`] = `
<div
  class="euiOverlayMask euiOverlayMask--aboveHeader"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal"
      style="max-width: 600px;"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
        type="button"
      >
        EuiIconMock
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            Configuration method
          </div>
        </div>
        <div
          class="euiModalBody"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--directionColumn euiFlexGroup--responsive"
              style="margin: -4px;"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiText euiText--small"
                  style="margin-top: 0px;"
                >
                  Choose how you would like to 
                  modify
                   your policy, either using a visual editor or writing JSON.
                </div>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <div
                    class="euiFlexItem"
                  >
                    <div
                      class="euiPanel euiPanel--paddingMedium selected-radio-panel"
                    >
                      <div
                        class="euiFormRow"
                        id="some_html_id-row"
                      >
                        <div
                          class="euiFormRow__fieldWrapper"
                        >
                          <div
                            aria-describedby="some_html_id-help"
                            class="euiRadio"
                            data-test-subj="createPolicyModalVisualRadio"
                          >
                            <input
                              checked=""
                              class="euiRadio__input"
                              id="some_html_id"
                              type="radio"
                              value=""
                            />
                            <div
                              class="euiRadio__circle"
                            />
                            <label
                              class="euiRadio__label"
                              for="some_html_id"
                            >
                              Visual editor
                            </label>
                          </div>
                          <div
                            class="euiFormHelpText euiFormRow__text"
                            id="some_html_id-help"
                          >
                            Use the visual editor to update your policy
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiFlexItem"
                  >
                    <div
                      class="euiPanel euiPanel--paddingMedium"
                    >
                      <div
                        class="euiFormRow"
                        id="some_html_id-row"
                      >
                        <div
                          class="euiFormRow__fieldWrapper"
                        >
                          <div
                            aria-describedby="some_html_id-help"
                            class="euiRadio"
                            data-test-subj="createPolicyModalJsonRadio"
                          >
                            <input
                              class="euiRadio__input"
                              id="some_html_id"
                              type="radio"
                              value=""
                            />
                            <div
                              class="euiRadio__circle"
                            />
                            <label
                              class="euiRadio__label"
                              for="some_html_id"
                            >
                              JSON editor
                            </label>
                          </div>
                          <div
                            class="euiFormHelpText euiFormRow__text"
                            id="some_html_id-help"
                          >
                            Use the JSON editor to update your policy using JSON.
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--primary"
                data-test-subj="createPolicyModalCancelButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonEmpty__content"
                >
                  <span
                    class="euiButtonEmpty__text"
                  >
                    Cancel
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary euiButton--fill"
                data-test-subj="createPolicyModalContinueButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Continue
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
